<?xml version="1.0" ?>
<launch>
  <arg name="robot_ip"               default="172.168.0.2"/>
  <arg name="load_gripper"           default="true" />
  <arg name="use_interactive_marker" default="false" />
  <arg name="use_gripper_gui"        default="true" />
  <arg name="load_franka_control"    default="true" />
  <arg name="rviz"                  default="true" />
  <arg name="teleop_on_same_machine" default="false" />
  <arg name="replay" default="false" />
  <arg name="optitrack_on" default="true" />

<!--   <rosparam ns="move_to_start" file="$(find franka_interactive_controllers)/config/start_pose.yaml" /> -->
  <!-- Bringup franka_interactive_bringup.laucnh -->
  <group if="$(arg load_franka_control)">
    <include file="$(find franka_interactive_controllers)/launch/franka_interactive_bringup_minimal.launch" >
      <arg name="robot_ip" value="$(arg robot_ip)" />
      <arg name="load_gripper" value="$(arg load_gripper)" />
      <arg name="use_gripper_gui" value="$(arg use_gripper_gui)" />
      <arg name="bringup_rviz" value="$(arg rviz)" />
    </include>
  </group>

  <!-- Load desired controller-->
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="effort_joint_trajectory_controller"/>
  <rosparam  ns="cartesian_pose_impedance_controller" command="load" file="$(find franka_interactive_controllers)/config/pose_impedance_control_additional_params.yaml"/>

  <arg name="node_name_gripper_control"            default="gripper_control_default" />
  <arg name="config_gripper_control"          default="$(find franka_interactive_controllers)/config/$(arg node_name_gripper_control).yaml" />
  <node name="$(arg node_name_gripper_control)" pkg="franka_interactive_controllers" type="gripper_control_from_topic.py" output="screen" clear_params="true">
  </node>


<!--   evenntually load the teloperation stuff -->
    <group if="$(arg teleop_on_same_machine)">
    <include file="$(find data_recorder)/launch/teleop_default.launch" >
    </include>
    </group>

<!--   Initialize Optitrack, not needed for replay -->
    <group unless="$(arg optitrack_on)">
    <include file="$(find natnet_bridge)/launch/optitrack_panda_21_05_24.launch" >
    </include>
    </group>

<!--     Initialize the readout of the controller -->
    <group if="$(arg replay)">
    <include file="$(find mouse_readout_ros)/launch/test_node.launch" >
    <arg name="node_name" value="mouse_readout_replay_node" />
    </include>
    </group>
    <group unless="$(arg replay)">
    <include file="$(find mouse_readout_ros)/launch/test_node.launch" >
    <arg name="node_name" value="mouse_readout_node" />
    </include>
    </group>



<!--   <node name="move_to_start" pkg="franka_interactive_controllers" type="move_to_start_joints.py" output="screen" required="true" /> -->

  <!-- IF interactive marker= true: Use interactive marker to define desired pose for impedance (stiffness + damping compensation) controller 
  <group if="$(arg use_interactive_marker)">
    <node name="interactive_marker" pkg="franka_interactive_controllers" type="interactive_marker.py" required="true" output="screen">
      <param name="link_name" value="panda_link0" />
      <remap from="desired_pose" to="/cartesian_impedance_controller/desired_pose" />
    </node>
  </group> -->

</launch>
